### DISCUSSION TITLE
Icon for unchecked state for checkbox

### DISCUSSION THREAD

**Discussion by joncaspeco** [chakra-ui/chakra-ui#7053] [Q&A]
I've been looking around to see if anyone else have had this issue. According to my design provided from the ux/ui team i need to have an icon in the checkbox control during the unchecked state as well in the checked state. Can’t seem to find a way of achieving this. All help is appreciated!

**Comment by wrux** [comment]
Yea It's possible as Chakra parses `isChecked` into the custom icon, so you can just change the icon if it's checked or not.

You can try it out in the code tool in the docs [here](https://chakra-ui.com/docs/components/checkbox#icon).

If you paste in my example below, I changed the custom icon component to change the fill between red/pink and also changed the path code based on `isChecked`.

```js
function CustomIcon(props) {
  const { isChecked, ...rest } = props

  const d = isChecked
    ? 'M12,0A12,12,0,1,0,24,12,12.013,12.013,0,0,0,12,0Zm0,19a1.5,1.5,0,1,1,1.5-1.5A1.5,1.5,0,0,1,12,19Zm1.6-6.08a1,1,0,0,0-.6.917,1,1,0,1,1-2,0,3,3,0,0,1,1.8-2.75A2,2,0,1,0,10,9.255a1,1,0,1,1-2,0,4,4,0,1,1,5.6,3.666Z'
    : 'M0,12a1.5,1.5,0,0,0,1.5,1.5h8.75a.25.25,0,0,1,.25.25V22.5a1.5,1.5,0,0,0,3,0V13.75a.25.25,0,0,1,.25-.25H22.5a1.5,1.5,0,0,0,0-3H13.75a.25.25,0,0,1-.25-.25V1.5a1.5,1.5,0,0,0-3,0v8.75a.25.25,0,0,1-.25.25H1.5A1.5,1.5,0,0,0,0,12Z'

  return (
    <Icon viewBox='0 0 24 24' {...rest}>
      <path fill={isChecked ? 'red':'pink'} d={d} />
    </Icon>
  )
}

function CustomCheckbox() {
  return (
    <Checkbox icon={<CustomIcon />} colorScheme='cyan'>
      Hello world
    </Checkbox>
  )
}

render(<CustomCheckbox />)
```
